### PR TITLE
Allow users to specify gocat extensions to install

### PR DESCRIPTION
**Pull request by uruwhy**
Instead of having sandcat gocat extensions included by default, users can specify which extensions they want included when dynamically compiling sandcat.  This can be done by adding in an HTTP header "gocat-extensions", whose value is a comma-separated list of extension names.

### CODE REVIEW

File: app/sand_svc.py [hunk 1]
```diff
@@ -18,16 +18,27 @@ def __init__(self, services):
         self.sandcat_dir = os.path.relpath(os.path.join('plugins', 'sandcat'))
 
     async def dynamically_compile_executable(self, headers):
-        name, platform = headers.get('file'), headers.get('platform')
+        # HTTP headers will specify the file name, platform, and comma-separated list of extension modules to include.
+        name, platform, extension_list_str = headers.get('file'), headers.get('platform'), headers.get('gocat-extensions', None)
```
Comment (privateducky) [inline app/sand_svc.py]: you can handle the extension_list_str -> extension_names in less code

File: app/sand_svc.py [hunk 2]
```diff
@@ -18,16 +18,27 @@ def __init__(self, services):
...
-                                          output_name=name)
+                                          output_name=name,
+                                          extension_names=extension_names)
         return await self.app_svc.retrieve_compiled_file(name, platform)
 
     async def dynamically_compile_library(self, headers):
-        name, platform = headers.get('file'), headers.get('platform')
+        # HTTP headers will specify the file name, platform, and comma-separated list of extension modules to include.
+        name, platform, extension_list_str = headers.get('file'), headers.get('platform'), headers.get('gocat-extensions', None)
+        extension_names = []
```
Comment (privateducky) [inline app/sand_svc.py]: same as above

File: app/sand_svc.py (hunk 1 above)
Comment (uruwhy) [reply app/sand_svc.py]: made it into a one-liner

File: app/sand_svc.py (hunk 2 above)
Comment (uruwhy) [reply app/sand_svc.py]: made it into a one-liner

**Pull request merged**